configuration/src/main/java/io/camunda/configuration/DocumentBasedHistory.java

```diff
@@ -13,17 +13,54 @@
 
 public class DocumentBasedHistory {
 
+  public static final int DEFAULT_HISTORY_MAX_DELAY_BETWEEN_RUNS = 60000;
+  private static final int DEFAULT_HISTORY_DELAY_BETWEEN_RUNS = 2000;
   private static final boolean DEFAULT_HISTORY_PROCESS_INSTANCE_ENABLED = true;
-
+  private static final String DEFAULT_HISTORY_ELS_ROLLOVER_DATE_FORMAT = "date";
+  private static final String DEFAULT_HISTORY_ROLLOVER_INTERVAL = "1d";
+  private static final int DEFAULT_HISTORY_ROLLOVER_BATCH_SIZE = 100;
+  private static final String DEFAULT_HISTORY_WAIT_PERIOD_BEFORE_ARCHIVING = "1h";
   private static final Map<String, String> LEGACY_BROKER_PROPERTIES =
       Map.of(
           "process-instance-enabled",
-          "zeebe.broker.exporters.camundaexporter.args.history.process-instance-enabled");
-
+          "zeebe.broker.exporters.camundaexporter.args.history.process-instance-enabled",
+          "els-rollover-date-format",
+          "zeebe.broker.exporters.camundaexporter.args.history.elsRolloverDateFormat",
+          "rollover-interval",
+          "zeebe.broker.exporters.camundaexporter.args.history.rolloverInterval",
+          "rollover-batch-size",
+          "zeebe.broker.exporters.camundaexporter.args.history.rolloverBatchSize",
+          "wait-period-before-archiving",
+          "zeebe.broker.exporters.camundaexporter.args.history.waitPeriodBeforeArchiving",
+          "delay-between-runs",
+          "zeebe.broker.exporters.camundaexporter.args.history.delayBetweenRuns",
+          "max-delay-between-runs",
+          "zeebe.broker.exporters.camundaexporter.args.history.maxDelayBetweenRuns");
   private final String prefix;
 
   private boolean processInstanceEnabled = DEFAULT_HISTORY_PROCESS_INSTANCE_ENABLED;
 
+  /** Date format for historical indices in Java DateTimeFormatter syntax */
+  private String elsRolloverDateFormat = DEFAULT_HISTORY_ELS_ROLLOVER_DATE_FORMAT;
+
+  /** Time range for creating dated indices (e.g., 1d creates daily indices). */
+  private String rolloverInterval = DEFAULT_HISTORY_ROLLOVER_INTERVAL;
+
+  /** Maximum number of process instances per archiving batch */
+  private int rolloverBatchSize = DEFAULT_HISTORY_ROLLOVER_BATCH_SIZE;
+
+  /**
+   * Grace period before archiving completed processes. Processes finished within this window are
+   * not yet archived.
+   */
+  private String waitPeriodBeforeArchiving = DEFAULT_HISTORY_WAIT_PERIOD_BEFORE_ARCHIVING;
+
+  /** Millisecond interval between archiver runs */
+  private int delayBetweenRuns = DEFAULT_HISTORY_DELAY_BETWEEN_RUNS;
```

Comment:
Just a thought: maybe we can make it a `Duration` but don't mind if you keep it as-is.